contracts/staking.tmy/src/staking.tmy.cpp

```diff
@@ -139,14 +139,22 @@ namespace stakingtoken
       settings_table_instance.set(settings, get_self());
    }
 
+   void stakingToken::_releasetoken(name staker, eosio::asset tokens_staked) {
```

Comment:
all the following can be in this function as it happens the same in both places.

suggest changing this function to accept the settings object, and the iterator to the staking allocation

```c++
            settings.total_releasing -= itr->tokens_staked;
            settings_table_instance.set(settings, get_self());

            staking_allocations_table.erase(itr);
            settings_table_instance.set(settings, get_self());  
            ```